app/controllers/api/v1/fees_controller.rb

```diff
@@ -4,17 +4,74 @@ module Api
   module V1
     class FeesController < Api::BaseController
       def show
-        # NOTE: instant fees might not be linked to any invoice, but add_on fees does not have any subscriptions
-        #       so we need a bit of logic to find the fee in the right organization scope
-        fee = Fee.left_joins(:invoice)
-          .left_joins(subscription: :customer)
-          .where('COALESCE(invoices.organization_id, customers.organization_id) = ?', current_organization.id)
+        fee = Fee.from_organization(current_organization)
           .find_by(id: params[:id])
 
         return not_found_error(resource: 'fee') unless fee
 
         render(json: ::V1::FeeSerializer.new(fee, root_name: 'fee'))
       end
+
+      def update
+        fee = Fee.from_organization(current_organization)
+          .find_by(id: params[:id])
+        result = Fees::UpdateService.call(fee:, params: update_params)
+
+        if result.success?
+          render(json: ::V1::FeeSerializer.new(fee, root_name: 'fee'))
+        else
+          render_error_response(result)
+        end
+      end
+
+      def index
+        result = FeesQuery.call(
+          organization: current_organization,
+          pagination: BaseQuery::Pagination.new(
+            page: params[:page],
+            limit: params[:per_page] || PER_PAGE,
+          ),
+          filters: BaseQuery::Filters.new(index_filters),
+        )
+
+        if result.success?
+          render(
+            json: ::CollectionSerializer.new(
+              result.fees,
+              ::V1::FeeSerializer,
+              collection_name: 'fees',
+              meta: pagination_metadata(result.fees),
+            ),
+          )
+        else
+          render_error_response(result)
+        end
+      end
+
+      private
+
+      def update_params
+        params.require(:fee).permit(:payment_status)
+      end
+
+      def index_filters
+        params.permit(
+          :fee_type,
+          :payment_status,
+          :external_subscription_id,
+          :external_customer_id,
+          :billable_metric_code,
+          :currency,
+          :created_at_from,
+          :created_at_to,
+          :failed_at_from,
+          :failed_at_to,
+          :succeeded_at_from,
+          :succeeded_at_to,
+          :refunded_at_from,
+          :refunded_at_to,
```

Comment:
Are we sure we really need to have these filters?

I mean, we can use the combination of `payment_status` and `created_at`, no?

Reply:
Yep it's one of the request of some customers. they want to be able to retrive fees based on their payment|success/failure/refund date, and not only the creation date